tests/rpmbuild.at

```diff
@@ -131,6 +131,37 @@ runroot rpmbuild -bs /data/SPECS/specstep.spec 2>&1|grep ^Executing|cut -d: -f1
 [])
 AT_CLEANUP
 
+# ------------------------------
+# Check if rpmbuild -ba *.spec works after 2038
+AT_SETUP([rpmbuild -ba *.spec after 2038])
+AT_KEYWORDS([build])
+RPMDB_INIT
+AT_CHECK([[
+touch -t 204205042355 ${RPMTEST}/data/SPECS/hello.spec
+touch -t 204205042355 ${RPMTEST}/data/SOURCES/hello-1.0.tar.gz
+touch -t 204205042355 ${RPMTEST}/data/SOURCES/hello-1.0-modernize.patch
+
+unset SOURCE_DATE_EPOCH
+run rpmbuild --quiet \
+  -ba  "${RPMTEST}"/data/SPECS/hello.spec
+
+run rpm -qp --qf '[%{FILENAMES} %{FILEMTIMES:date}\n]' "${RPMTEST}"/build/SRPMS/hello-1.0-1.src.rpm
+run rpm -qp --qf '[%{FILENAMES} %{FILEMTIMES:date}\n]' "${RPMTEST}"/build/RPMS/*/hello-1.0-1.*.rpm | grep FAQ
```

Comment:
Put different tests into separate `AT_CHECK()` sections (there can be multiple under one `AT_SETUP` case), it makes tests easier to read, write and understand. Always use `runroot` if at all possible - rpmbuild is one of the exceptions where you might have to settle for `run` because compilers and the like are not available in the root.